### ISSUE TITLE
Failed to perform DigitalOcean request

### ISSUE THREAD

**Issue by jinrenjie** [Jigsaw-Code/outline-server#466]
Error report：

```json
{
	"environment": "production",
	"extra": {
		"crashed_process": "renderer[1]",
		"crashed_url": "outline://web_app/index.html?version=1.2.2&sentryDsn=https%3A%2F%2F533e56d1b2d64314bd6092a574e6d0f1%40sentry.io%2F215496",
		"session:duration": 1215983
	},
	"release": "1.2.2",
	"user": {
		"ip_address": "{{auto}}"
	},
	"event_id": "52cacb3166ff4cae91836accf8415500",
	"exception": {
		"mechanism": {
			"handled": true,
			"type": "generic"
		},
		"values": [{
			"stacktrace": {
				"frames": [{
					"colno": 21,
					"filename": "app:///node_modules/@sentry/browser/dist/index.js",
					"function": "XMLHttpRequest.wrapped",
					"in_app": true,
					"lineno": 2471
				}]
			},
			"type": "XhrError",
			"value": ""
		}]
	},
	"logger": "javascript",
	"platform": "javascript",
	"project": "215496",
	"request": {
		"headers": {},
		"url": "outline://web_app/index.html?version=1.2.2&sentryDsn=https%3A%2F%2F533e56d1b2d64314bd6092a574e6d0f1%40sentry.io%2F215496"
	},
	"sdk": {
		"name": "sentry.javascript.electron",
		"packages": [{
			"name": "npm:@sentry/browser",
			"version": "4.0.0-beta.12"
		}, {
			"name": "npm:@sentry/node",
			"version": "4.0.0-beta.12"
		}, {
			"name": "npm:@sentry/node",
			"version": "0.8.1"
		}],
		"version": "0.8.1"
	},
	"transaction": "/Applications/Outline Manager.app/Contents/Resources/app.asar/node_modules/@sentry/browser/dist/index.js",
	"trimHeadFrames": 0,
	"tags": {
		"event_type": "javascript"
	},
	"breadcrumbs": [{
		"timestamp": 1565957976.144,
		"category": "ui.click",
		"message": " > outline-step-view > div.container > div#digital-ocean.card > div.card-footer"
	}, {
		"timestamp": 1565957976.144,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957976.144,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957976.193,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957976.194,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957976.353,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957976.354,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957976.438,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565957976.438,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957976.438,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957976.615,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957976.615,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957976.627,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565957976.627,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957976.628,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957976.719,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957976.719,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957976.841,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565957976.841,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957976.841,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957977.019,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957977.02,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957977.023,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565957977.023,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957977.023,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957977.052,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957977.052,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957977.179,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565957977.179,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957977.179,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957977.347,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565957977.348,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957977.348,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957977.461,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957977.461,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957977.463,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957977.463,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957977.494,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565957977.495,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957977.495,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957978.013,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957978.013,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957978.106,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957978.106,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957978.396,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957978.397,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957978.467,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957978.468,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957978.897,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957978.897,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957978.957,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957978.958,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957979.266,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957979.266,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957979.523,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957979.524,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957979.64,
		"category": "electron",
		"message": "app.browser-window-blur",
		"type": "ui"
	}, {
		"timestamp": 1565957979.903,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957979.904,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957979.918,
		"category": "electron",
		"message": "app.browser-window-focus",
		"type": "ui"
	}, {
		"timestamp": 1565957979.904,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957979.907,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957980.266,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957980.266,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957980.296,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957980.297,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957980.661,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957980.661,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957980.666,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957980.667,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957981.013,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957981.013,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957981.448,
		"category": "electron",
		"message": "app.browser-window-blur",
		"type": "ui"
	}, {
		"timestamp": 1565957984.379,
		"category": "electron",
		"message": "app.browser-window-focus",
		"type": "ui"
	}, {
		"timestamp": 1565957984.941,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565957984.941,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565957984.942,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565957985.542,
		"category": "electron",
		"message": "app.browser-window-blur",
		"type": "ui"
	}, {
		"timestamp": 1565957985.545,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565957985.545,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565957985.561,
		"category": "electron",
		"message": "app.browser-window-focus",
		"type": "ui"
	}, {
		"timestamp": 1565957986.216,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565957986.216,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}, {
		"timestamp": 1565957986.524,
		"category": "electron",
		"message": "app.browser-window-blur",
		"type": "ui"
	}, {
		"timestamp": 1565957987.857,
		"category": "electron",
		"message": "app.browser-window-focus",
		"type": "ui"
	}, {
		"timestamp": 1565957988.505,
		"category": "electron",
		"message": "app.browser-window-blur",
		"type": "ui"
	}, {
		"timestamp": 1565957997.094,
		"category": "electron",
		"message": "app.browser-window-focus",
		"type": "ui"
	}, {
		"timestamp": 1565957997.874,
		"category": "electron",
		"message": "app.browser-window-blur",
		"type": "ui"
	}, {
		"timestamp": 1565957999.728,
		"category": "electron",
		"message": "app.browser-window-focus",
		"type": "ui"
	}, {
		"timestamp": 1565958000.52,
		"category": "electron",
		"message": "app.browser-window-blur",
		"type": "ui"
	}, {
		"timestamp": 1565958001.873,
		"category": "electron",
		"message": "app.browser-window-focus",
		"type": "ui"
	}, {
		"timestamp": 1565958002.719,
		"category": "ui.click",
		"message": "div.container > div#digital-ocean.card > div.card-footer > paper-button.primary"
	}, {
		"timestamp": 1565958002.72,
		"category": "console",
		"level": "info",
		"message": "found access token in local storage"
	}, {
		"timestamp": 1565958002.72,
		"category": "console",
		"level": "info",
		"message": "Requesting account"
	}, {
		"timestamp": 1565958004.061,
		"category": "electron",
		"message": "app.browser-window-blur",
		"type": "ui"
	}, {
		"timestamp": 1565958004.265,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 200,
			"url": "https://api.digitalocean.com/v2/account/"
		},
		"type": "http"
	}, {
		"timestamp": 1565958004.279,
		"category": "electron",
		"message": "app.browser-window-focus",
		"type": "ui"
	}, {
		"timestamp": 1565958004.265,
		"category": "console",
		"level": "info",
		"message": "Requesting droplet by tag"
	}, {
		"timestamp": 1565958004.656,
		"category": "xhr",
		"data": {
			"method": "GET",
			"status_code": 0,
			"url": "https://api.digitalocean.com/v2/droplets/?tag_name=shadowbox"
		},
		"type": "http"
	}, {
		"timestamp": 1565958004.656,
		"category": "console",
		"level": "error",
		"message": "Failed to perform DigitalOcean request"
	}],
	"contexts": {
		"app": {
			"app_name": "Outline Manager",
			"app_version": "1.2.2"
		},
		"chrome": {
			"name": "Chrome",
			"type": "runtime",
			"version": "61.0.3163.100"
		},
		"device": {
			"arch": "x64"
		},
		"node": {
			"name": "Node",
			"type": "runtime",
			"version": "8.9.3"
		},
		"os": {
			"kernel_version": "18.7.0"
		},
		"runtime": {
			"name": "Electron",
			"version": "2.0.16"
		}
	}
}
```
<img width="1360" alt="WX20190816-203838@2x" src="https://user-images.githubusercontent.com/5665135/63168145-ec49cc00-c065-11e9-9724-1457dca41ba2.png">

**Comment by spotty118** [comment]
I'm having the same exact issue.

**Comment by jinrenjie** [comment]
@spotty118 I have submitted the pull request, DigitalOcean not allow CORS

**Comment by alalamav** [comment]
It seems like DigitalOcean changed their API configuration to disallow CORS. Could be related to the DigitalOcean [last](https://status.digitalocean.com/incidents/np0bfb0bk79h) [incidents](https://status.digitalocean.com/incidents/3k09rrjm9nmc). We started seeing this failure two days ago.

@jinrenjie, thank you for the PR. I'm not sure that we want to disable same-origin policy altogether. Let's see if we can find another fix, else I'll approve yours.

**Comment by alalamav** [comment]
The fix for this issue will go out today in the Outline Manager v1.2.3.

**Comment by nitrohorse** [comment]
> The fix for this issue will go out today in the Outline Manager v1.2.3.

Still experiencing this issue :confused: Nothing appears to happen after selecting "Create Server."

**Comment by alalamav** [comment]
Hi @nitrohorse, thanks for letting us know. It seems like it's a different CORS issue on v1.2.3. I'm seeing that `OPTIONS` requests are returning 403. I'll look into it as soon as possible.

**Comment by z764969689** [comment]
The issue still exists on Windows 10 so far. Hopefully, it would be fixed asap.

**Comment by alalamav** [comment]
This issue has been fixed in #476 and will be released in v1.2.4 later today.

**Comment by nitrohorse** [comment]
^^ Nice! Confirmed it's working for me now :smile: thank you!
